modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ErlangClientCodegen.java

```diff
@@ -0,0 +1,393 @@
+
+        // e.g. PetApi.erl => pet_api.erl
+        return this.packageName + "_" + underscore(name) + "_api";
+    }
+
+    @Override
+    public String toOperationId(String operationId) {
+        // method name cannot use reserved keyword, e.g. return
+        if (isReservedWord(operationId)) {
+            LOGGER.warn(operationId + " (reserved word) cannot be used as method name. Renamed to " + underscore(sanitizeName("call_" + operationId)));
+            operationId = "call_" + operationId;
+        }
+
+        return underscore(operationId);
+    }
+
+    @Override
+    public Map<String, Object> postProcessOperations(Map<String, Object> objs) {
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        List<CodegenOperation> os = (List<CodegenOperation>) operations.get("operation");
+        List<ExtendedCodegenOperation> newOs = new ArrayList<ExtendedCodegenOperation>();
+        Pattern pattern = Pattern.compile("(.*)\\{([^\\}]+)\\}(.*)");
+        for (CodegenOperation o : os) {
+            // force http method to lower case
+            o.httpMethod = o.httpMethod.toLowerCase();
+
+            if (o.isListContainer) {
+                o.returnType = "[" + o.returnBaseType + "]";
+            }
+
+            ArrayList<String> pathTemplateNames = new ArrayList<String>();
+            Matcher matcher = pattern.matcher(o.path);
+            StringBuffer buffer = new StringBuffer();
+            while (matcher.find()) {
+                String pathTemplateName = matcher.group(2);
+                matcher.appendReplacement(buffer, "$1" + "\", " + camelize(pathTemplateName) + ", \"" + "$3");
+                pathTemplateNames.add(pathTemplateName);
+            }
+            ExtendedCodegenOperation eco = new ExtendedCodegenOperation(o);
+            if (buffer.toString().isEmpty()) {
+                eco.setReplacedPathName(o.path);
+            } else {
+                eco.setReplacedPathName(buffer.toString());
+            }
+            eco.setPathTemplateNames(pathTemplateNames);
+            newOs.add(eco);
+        }
+        operations.put("operation", newOs);
+        return objs;
+    }
+
+    public void setPackageName(String packageName) {
+        this.packageName = packageName;
+    }
+
+    public void setPackageVersion(String packageVersion) {
+        this.packageVersion = packageVersion;
+    }
+
+    String length(Object o) {
+        return Integer.toString((((ExtendedCodegenOperation) o).allParams).size());
+    }
+
+    @Override
+    public String escapeQuotationMark(String input) {
+        // remove " to avoid code injection
+        return input.replace("\"", "");
+    }
+
+    @Override
+    public String escapeUnsafeCharacters(String input) {
+        return input.replace("*/", "*_/").replace("/*", "/_*");
+    }
+
+    class ExtendedCodegenOperation extends CodegenOperation {
```

Comment:
@tsloughter Do we still need this Extended class after using vendor extensions to stored the customized path?

Reply:
Oh, yea, probably. Will clean it up. More might have got left like that because of how I copied the module from another implementation.

Reply:
Or no, not so simple, would have to look closer at how `ExtendedCodegenOperation` is used in `postProcessOperations` because that is still needed, maybe it can just be replaced with `CodegenOperation` but just doing that naively and running it didn't compile, so I'll need to look more later.

Reply:
@tsloughter that's ok. Let's keep it for now.